### PR TITLE
Fix BatchNorm backward synchronization

### PR DESCRIPTION
**Pull request by anko-intel**
## Description ##
Fix the issue #18610 - synchronization problem with running variables for backward pass of BatchNorm

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change`

### CODE REVIEW

File: tests/python/unittest/test_gluon.py [hunk 1]
```diff
@@ -665,6 +665,34 @@ def transpose(shape):
...
+        layer = nn.BatchNorm()
+        layer.initialize(ctx=ctx)
+        for _ in range(3):
+            data = mx.nd.random.normal(loc=10, scale=2, shape=(1, 3, 10, 10), ctx=ctx)
+            with mx.autograd.record():
+                out = layer(data)
+            out.backward()
+
+        # check if each read give the same value
+        var1 = eval(read_op)
```
Comment (wkcn) [inline tests/python/unittest/test_gluon.py:689]: Thank you for the contribution!
For safety, I suggest to use `var1 = getattr(layer, variable).data().asnumpy()`.

File: tests/python/unittest/test_gluon.py [hunk 2]
```diff
@@ -665,6 +665,34 @@ def transpose(shape):
...
+        for _ in range(10):
+            var2 = eval(read_op)
```
Comment (wkcn) [inline tests/python/unittest/test_gluon.py:691]: `var2 = getattr(layer, variable).data().asnumpy()`.

File: tests/python/unittest/test_gluon.py [hunk 3]
```diff
@@ -665,6 +665,34 @@ def transpose(shape):
...
 
+@with_seed()
+@pytest.mark.parametrize('cudnn_off', [True, False])
+@pytest.mark.parametrize('variable', ['running_var', 'running_mean'])
+def test_batchnorm_backward_synchronization(cudnn_off, variable):
+    """
+    Tests if synchronization of BatchNorm running variables is done correctly.
+    If not, the test sometimes fails - depending on the timing.
+    """
+    ctx = mx.cpu() if cudnn_off else mx.gpu()
```
Comment (wkcn) [inline tests/python/unittest/test_gluon.py:676]: It should be `ctx=default_context()`.

In CI, there are tests on CPU and GPU. We don't need to specify the context.

Comment (anko-intel) [reply tests/python/unittest/test_gluon.py:676]: Thanks for the tip @wkcn,  
Is there any command line option or environment value to enforce particular context  when I run pytest locally?

File: tests/python/unittest/test_gluon.py (hunk 1 above)
Comment (anko-intel) [reply tests/python/unittest/test_gluon.py:689]: Thanks!

**Comment by anko-intel** [comment]
@mxnet-bot [codecov/project]

File: tests/python/unittest/test_gluon.py (hunk 3 above)
Comment (wkcn) [reply tests/python/unittest/test_gluon.py]: There seems to no command line to specify the context.
The default context is CPU.
We can create a python file, like https://github.com/apache/incubator-mxnet/blob/master/tests/python/gpu/test_gluon_gpu.py

'''
import mxnet ad mx
from some_test_file import *
set_default_context(mx.gpu(0))
'''
It will switch to gpu context.

**Comment by TaoLv** [comment]
Thanks for the contribution, @anko-intel. Does the problem also exist on v1.x? If yes, could you please cherry pick the fix to the branch also?
